File: Examples/Node.js/SDK/Text/FindTextWorksheet.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var CellsApi = require('asposecellscloud');
var configProps = require('../Config/config.json');
var data_path = '../../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Cells API SDK
var cellsApi = new CellsApi(config);

// Set input file name
var name = "Sample_Test_Book.xls";
var sheetName = "Sheet2";
var text = "aspose";
try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');
	
	// Invoke Aspose.Cells Cloud SDK API to find text in a worksheet
	cellsApi.PostWorkSheetTextSearch(name, sheetName, text, null, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			responseMessage.body.TextItems.TextItemList.forEach(function(item) {
				console.log("Text: " + item.Text);
				console.log("Href: " + item.link.Href);
				console.log("--");
				});
		});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Java/SDK/src/main/java/com/aspose/cells/cloud/examples/text/FindTextWorksheet.java
package com.aspose.cells.cloud.examples.text;

import com.aspose.cells.api.CellsApi;
import com.aspose.cells.cloud.examples.Configuration;
import com.aspose.cells.cloud.examples.Utils;
import com.aspose.storage.api.StorageApi;

import java.io.IOException;
import java.nio.file.Path;

public class FindTextWorksheet {

    public static void main(String... args) throws IOException {
        // ExStart: find-text-worksheet

        try {
            // Instantiate Aspose Storage API SDK
            StorageApi storageApi = new StorageApi(Configuration.apiKey, Configuration.appSID, true);

            // Instantiate Aspose Words API SDK
            CellsApi cellsApi = new CellsApi(Configuration.apiKey, Configuration.appSID, true);
            String input = "sample1.xlsx";
            Path inputFile = Utils.getPath(FindTextWorksheet.class, input);

            storageApi.PutCreate(input, null, null, inputFile.toFile());
            String text = "aspose";
            String sheetName = "Sheet1";

            com.aspose.cells.model.TextItemsResponse apiResponse = cellsApi.PostWorkSheetTextSearch(input, sheetName,
                    text, Utils.STORAGE, Utils.FOLDER);

            for (com.aspose.cells.model.TextItem item : apiResponse.getTextItems().getTextItemList()) {
                System.out.println("Text: " + item.getText());
                System.out.println("Href: " + item.getLink().getHref());
                System.out.println("");
            }
        }

        catch (Exception e) {
            e.printStackTrace();
        }
        // ExEnd: find-text-worksheet
    }

}